### PR TITLE
add tests for container kill and exists

### PR DESCRIPTION
**Pull request by lsm5**
depends on https://github.com/containers/libpod/pull/5374 . Also, tests are currently failing due to a crun issue being worked on separately.

### CODE REVIEW

**Comment by rhatdan** [comment]
@lsm5 needs a rebase

File: pkg/bindings/test/containers_test.go [hunk 1]
```diff
@@ -380,4 +380,103 @@ var _ = Describe("Podman containers ", func() {
...
+	It("podman kill a running container by ID with TERM", func() {
+		// Killing a running container by ID with TERM should work
+		var name = "top"
+		cid, err := bt.RunTopContainer(&name, &bindings.PFalse, nil)
+		Expect(err).To(BeNil())
+		err = containers.Kill(bt.conn, cid, "SIGTERM")
+		Expect(err).To(BeNil())
+	})
+
+	// This test panics after a 10 minute timeout.
```
Comment (lsm5) [inline pkg/bindings/test/containers_test.go]: @baude PTAL ^

Comment (lsm5) [reply pkg/bindings/test/containers_test.go:486]: @baude re-ping

**Comment by lsm5** [comment]
@baude @mheon all tests passed. PTAL..

**Comment by rhatdan** [comment]
LGTM
@mheon @baude @vrothberg @giuseppe @edsantiago @cevich PTAL

**Comment by lsm5** [comment]
> Is it really necessary to use SIGKILL so often? I was trained to use that only as a last resort. Maybe you could alternate SIGINT and SIGTERM?
> 
> Also, what would you think of adding an `Exists()` test after one or two of the kill tests? E.g. after line 483? See, the thing is, you're not really testing kill: you're testing "kill does not give me an error". Not the same thing.

@edsantiago addressed both comments. PTAL..

**Comment by edsantiago** [comment]
LGTM assuming tests pass. Thank you!

**Comment by edsantiago** [comment]
Test failure in [rootless local](https://storage.googleapis.com/cirrus-ci-5385732420009984-fcae48/artifacts/containers/libpod/5381542914555904/html/system_test.log.html):
```
Error: error removing container b52d9a50bceff47592a01cde287cf54a3f072570f045998efdcb0ee507388803 root filesystem: 1 error occurred:
 	* unlinkat /home/some18944dude/.local/share/containers/storage/overlay/428c1faaae3530ee1f58f0e7c55f6e45c0eb76580e75abc9bbc08cf2e28bc8d5/merged: device or resource busy
```
I'm guessing it's a flake but once again it's not one I've seen before. Could someone please investigate?

**Comment by edsantiago** [comment]
/lgtm

...but I would really like someone mount-knowledgeable (@mheon? @giuseppe?) to look at the flake above and help me understand how to address it / prevent it in the future (or, if it's a real podman problem, which I kind of think it might be, how to gather more data).

**Comment by mheon** [comment]
It seems like the mountpoint is still active when we try and remove it. Hard to tell exactly what's going on without having a look at what exactly is mounted when we try and remove - could be overlay failing to unmount, could be our SHM mount?

**Comment by edsantiago** [comment]
All of those are such simple tests: `podman run X`. Unfortunately I don't have a way of knowing which of those containers was the one leaving the mount behind. Would it help if I reworked the test such that it does 'podman rm` immediately after each `run` command?

**Comment by lsm5** [comment]
@edsantiago everything passes now, so maybe we're good to go?

**Comment by edsantiago** [comment]
I gave it my /lgtm blessing but I don't have privs to `/approve`

**Pull request merged**